analyses/snv-callers/util/wrangle_functions.R

```diff
@@ -0,0 +1,260 @@
+  wxs_maf_granges <- GenomicRanges::GRanges(
+    seqnames = maf_wxs$Chromosome,
+    ranges = IRanges::IRanges(
+      start = maf_wxs$Start_Position,
+      end = maf_wxs$End_Position
+    )
+  )
+
+  # Find the overlap of the BED regions and the mutations.
+  overlap <- GenomicRanges::findOverlaps(wxs_maf_granges, wxs_bed_granges,
+    maxgap = bp_window
+  )
+
+  # Calculate of ratio of variants in this BED
+  ratio <- length(overlap@from) / nrow(maf_wxs)
+
+  # What fraction of mutations are in these bed regions?
+  cat(
+    "Ratio of variants in this BED:", ratio,
+    "Ratio of variants being filtered out:", 1 - ratio
+  )
+
+  # Only keep those in the BED regions
+  maf_wxs <- maf_wxs[unique(overlap@from), ]
+
+  # Tack these back on to the WGS samples
+  filt_maf_df <- maf_df %>%
+    dplyr::filter(experimental_strategy == "WGS") %>%
+    dplyr::bind_rows(maf_wxs)
+
+  # Return this filtered matrix
+  return(filt_maf_df)
+}
+
+calculate_tmb <- function(maf_df, wgs_size, wxs_size) {
+  # Calculate Tumor Mutational Burden for each sample given their WGS or WXS
+  # sizes in bp. Ideally have filtered the WXS mutations previously using the
+  # `wxs_bed_filter` function.
+  #
+  # TMB = # variants / size of the genome or exome surveyed
+  #
+  # Args:
+  #   maf_df: maf data.frame that has been turned into a data.frame and has
+  #           WXS mutations filtered using `wxs_bed_filter` function.
+  #   wgs_size: genome size in bp to be used for WGS samples
+  #   wxs_size: genome size in bp to be used for WGS samples
+  #
+  # Make a genome size variable
+  tmb <- maf_df %>%
+    dplyr::mutate(genome_size = dplyr::recode(experimental_strategy,
+      "WGS" = wgs_size,
+      "WXS" = wxs_size
+    )) %>%
+    dplyr::group_by(
+      Tumor_Sample_Barcode, experimental_strategy, genome_size,
+      short_histology
+    ) %>%
+    # Count number of mutations for that sample
+    dplyr::summarize(mutation_count = dplyr::n()) %>%
+    # Calculate TMB
+    dplyr::mutate(tmb = mutation_count / (genome_size / 1000000))
+
+  return(tmb)
+}
+
+annotr_maf <- function(maf_df, annotation_obj = annotations, bp_window = 0) {
+  # Annotate the genomic regions mutations are in from a MAF formatted data.
+  # frame with AnnotatR object.
+  #
+  # Args:
+  #   maf_df: maf data that has been turned into a data.frame.
+  #   annotation_obj: an annotation object with the desired annotations.
+  #   bp_window: how many base pairs away can it be from the BED region to still
+  #              be included? Default is 0 bp. This argument gets forwarded
+  #              to GenomicRanges::findOverlaps's `maxgap` argument.
+  #
+  # Read in the genomic regions annotation object
+  annotations <- readr::read_rds(annot_rds)
+
+  # Use custom function to turn our MAF data into a GRanges
+  maf_ranges <- maf_to_granges(maf_df)
+
+  # Intersect the regions we read in with the annotations
+  annot_matches <- GenomicRanges::findOverlaps(annotation_obj, maf_ranges,
```

Comment:
```suggestion
  annot_matches <- GenomicRanges::findOverlaps(
    annotation_obj, 
    maf_ranges,
```

Reply:
Should this be `annotations` instead of `annotation_obj`?

Reply:
Yes. This whole argument handling is messed up. I'm changing it so `annotation_file` is for specifying a file.path to the annotation RDS.